Mathlib/Data/KVMap.lean

```diff
@@ -0,0 +1,23 @@
+/-
+Copyright (c) 2022 Alex J. Best. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex J. Best
+-/
+import Lean.Data.KVMap
+
+/-!
+# Additional functionality for `KVMap`
+-/
+
+namespace Lean.KVMap
+
+/-- erase pairs whose names match the second argument from a list of
+`Name × DataValue` pairs-/
+def eraseCore : List (Name × DataValue) → Name → List (Name × DataValue)
+  | l, n => List.dropWhile (fun a => a.1 == n) l
```

Comment:
`List.dropWhile` will only remove an initial segment. I think you just want `List.filter (fun a => a.1 \ne n)`.

Reply:
🤦 Thanks for catching this critical mistake Scott!